### ISSUE TITLE
Add InternetDomain & related?

### ISSUE THREAD

**Issue by Jamie-SA** [semanticarts/gist#66]
In the DCAO we need Internet domains (semanticarts.com) and sub-domains (either "www" or "www.semanticarts.com"). Maybe also port and protocol (such as http, https, sparql, ftp, smtp, ssh, ?).

Seems like these might belong in gistIoT.

**Comment by rjyounes** [comment]
Jamie has implemented in DCA for the time being because he needs it.

Consider adding these when we de-modularize gistIoT. See #323.

**Comment by uscholdm** [comment]
InternetDomain could be a subclass of Place, which subdivided into PhysicalPlace and Virtual (Cyber)PLace.

**Comment by uscholdm** [comment]
@Jamie-SA Can you have another look at this, it is deferred, should we close or activate?

**Comment by Jamie-SA** [comment]
This might end up in a sub-gist.
Either way, it probably should be considered in relation to several other issues related to addresses: #721, #594, etc.

**Comment by Jamie-SA** [comment]
With recent changes to gist, I think an Internet domain might now be a gist:ElectronicAddress with a gist:ElectronicAddressType.

**Comment by rjyounes** [comment]
I agree, this is a `gist:ElectronicAddress`. A sticking point is it can't use `gist:containedText`, which has range `xsd:string`, if we want to type the value as `xsd:anyURI`. I've defined a datatype property `:url` in client ontologist to get around this.

Closing this issue.

**Comment by Jamie-SA** [comment]
There is one aspect of an Internet Domain that is different, a domain is not an IRI as it does not have a protocol. So using xsd:anyURI would probably be incorrect. So I guess it should be a string.

When thought of in this light, would you think of it as a gist:ElectronicAddress?

At the moment I'm thinking it should not be, as it is not, by itself, a destination. The equivalent website URL (the domain prefixed with https://) would be a gist:ElectronicAddress.

**Comment by uscholdm** [comment]
> I agree, this is a `gist:ElectronicAddress`. A sticking point is it can't use `gist:containedText`, which has range `xsd:string`, if we want to type the value as `xsd:anyURI`. I've defined a datatype property `:url` in client ontologist to get around this.
> 
> Closing this issue.

The range could be union of string and anyUri.
